File: Form1.vb
Imports processPicker
Imports JFWAPICTRLLib
Public Class Form1
	Dim jfw As New JFWApi
	Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
		processPicker.runConsole("python", "-i")
		AddHandler processPicker.con.OutputDataReceived, AddressOf Process_OutputDataReceived
		AddHandler processPicker.con.ErrorDataReceived, AddressOf Process_errorDataReceived
	End Sub

	Private Sub Process_OutputDataReceived(sender As Object, e As DataReceivedEventArgs)
		If e.Data IsNot Nothing Then
			outputBox.AppendText(e.Data & Environment.NewLine)
			jfw.SayString(e.Data, 0)
		End If
	End Sub

	Private Sub Process_errorDataReceived(sender As Object, e As DataReceivedEventArgs)
		If e.Data IsNot Nothing Then
			outputBox.AppendText(e.Data & Environment.NewLine)
			My.Computer.Audio.Play("InconsistencyBuzzer.wav")
			jfw.SayString(e.Data, 0)
		End If
	End Sub

	Private Sub inputBox_KeyDown(sender As Object, e As KeyEventArgs) Handles inputBox.KeyDown
		If e.KeyCode = Keys.F6 Then
			outputBox.Focus()
		ElseIf e.KeyCode = Keys.Enter Then
			Dim lines() As String = inputBox.Lines
			Dim index As Integer = inputBox.GetLineFromCharIndex(inputBox.SelectionStart)
			Dim Input As String = inputBox.Lines(index)
			processPicker.con.StandardInput.WriteLine(Input)
			If Input <> "" Then
				outputBox.AppendText(Input & Environment.NewLine)
				If index <> lines.Length - 1 Then
					e.SuppressKeyPress = True
					inputBox.AppendText(Input & Environment.NewLine)
				End If
			Else
				e.SuppressKeyPress = True
			End If
		End If
	End Sub

	Private Sub Form1_FormClosing(sender As Object, e As FormClosingEventArgs) Handles MyBase.FormClosing
		processPicker.con.Kill()
		processPicker.con.Dispose()
	End Sub

	Private Sub saveItem_Click(sender As Object, e As EventArgs) Handles saveItem.Click
		Dim choice As New SaveFileDialog
		If choice.ShowDialog = DialogResult.OK Then System.IO.File.WriteAllText(choice.FileName, outputBox.Text)
	End Sub

	Private Sub exitItem_Click(sender As Object, e As EventArgs) Handles exitItem.Click
		Me.DestroyHandle()
	End Sub

	Private Sub copyInputItem_Click(sender As Object, e As EventArgs) Handles copyInputItem.Click

		If inputBox.SelectedText.Length = 0 Then
			inputBox.SelectAll()
			inputBox.Copy()
			inputBox.DeselectAll()
		Else
			inputBox.Copy()
		End If
	End Sub

	Private Sub switchConsole_Click(sender As Object, e As EventArgs) Handles switchConsole.Click
		switchConsoleDLG.ShowDialog()
	End Sub
End Class

File: AccessibleConsole/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using static System.Windows.Forms.VisualStyles.VisualStyleElement;
using ScreenReaderAPIWrapper;
using System.Media;

namespace AccessibleConsole
{
	public partial class Form1 : Form
	{
		public Form1()
		{
			InitializeComponent();
		}
		Process myProcess = new Process();
		ScreenReader ScreenReader = new ScreenReader();
		SoundPlayer error= new SoundPlayer("error.wav");
		SoundPlayer outPut = new SoundPlayer("feedback.wav");

		private void Form1_Load(object sender, EventArgs e){
			myProcess.StartInfo.FileName = "python";
			myProcess.StartInfo.Arguments = "-i";
			myProcess.StartInfo.UseShellExecute=false;
			myProcess.StartInfo.CreateNoWindow = true;
			myProcess.StartInfo.RedirectStandardInput=true;
			myProcess.StartInfo.RedirectStandardOutput = true;
			myProcess.StartInfo.RedirectStandardError = true;
			myProcess.OutputDataReceived += handleOutput;
			myProcess.ErrorDataReceived += handleErrors;
			myProcess.Start();
			myProcess.BeginOutputReadLine();
			myProcess.BeginErrorReadLine();
		}

		void handleOutput(object sender, DataReceivedEventArgs e){
            Invoke((MethodInvoker)(() => {
                outputBox.Text += e.Data + Environment.NewLine;
                ScreenReader.SayString(e.Data,false);
				outPut.Play();            }));        }

		void handleErrors(object sender, DataReceivedEventArgs e) {
			Invoke((MethodInvoker)(() => {
				outputBox.Text += e.Data + Environment.NewLine;
				ScreenReader.SayString(e.Data);
			error.Play();	}));		}

		private void Form1_FormClosing(object sender, FormClosingEventArgs e) { 
			myProcess.Dispose();		}

		private void inputBox_KeyPress(object sender, KeyPressEventArgs e)		{
			if (e.KeyChar == (Char)Keys.Enter) {
				int index = inputBox.GetLineFromCharIndex(inputBox.GetFirstCharIndexOfCurrentLine());
				string[] lines = inputBox.Lines;
				string input= lines[index];
					myProcess.StandardInput.WriteLine(input);
					ScreenReader.SayString(input);
                    if (index != lines.Length - 1)
					e.Handled = true;
                if (input != "")
                    outputBox.Text += input + Environment.NewLine;
            }
		}

        private void saveToolStripMenuItem_Click(object sender, EventArgs e)
        {
			SaveFileDialog saveFileDialog = new SaveFileDialog();
			if (saveFileDialog.ShowDialog() == DialogResult.OK)			{
				string path= saveFileDialog.FileName;
				System.IO.File.WriteAllText(path, outputBox.Text);
			}
        }

        private void exitToolStripMenuItem_Click(object sender, EventArgs e)
        {
			this.Close();
        }
    }
}
